themes/src/main/resources/theme/base/admin/resources/partials/user-detail.html

```diff
@@ -97,6 +97,42 @@
                 </div>
                 <kc-tooltip>{{:: 'email-verified.tooltip' | translate}}</kc-tooltip>
             </div>
+            <div class="form-group clearfix block" data-ng-if="create">
+                <label class="col-md-2 control-label" for="selected-groups">{{:: 'groups-joining' | translate}}</label>
+                <div class="col-sm-6">
+                    <table class="table table-striped table-bordered" style="margin-top: 0px" id="selected-groups">
+                        <thead>
+                        <tr>
+                            <th class="kc-table-actions" colspan="2">
+                                <div class="form-inline col-md-12" style="width: 107%">
+                                    <div class="form-group" style="width: 100%">
+                                        <div class="input-group" style="width: 100%">
+                                            <input type="hidden" ui-select2="groupsUiSelect" id="groups" data-ng-change="selectGroup(selectedGroup);" data-ng-model="selectedGroup" data-placeholder="{{:: 'groups-joining-select.placeholder' | translate}}..."/>
```

Comment:
I've copy-pasted a component used in Permissions UI:
![image](https://user-images.githubusercontent.com/6166832/93893134-970ece00-fced-11ea-9e6f-cfa4c838dd2c.png)

IMO, a tabular view better highlights groups hierarchy, compared to a "tags input".

Reply:
Yes, you're correct. But there it works a bit differently. Once you've clicked a policy in the search bar, it appears in the table below and the search bar is clears. There are no duplicit "remove" buttons.

Reply:
@vmuzikar The `x` means "clear the field".  That's different from remove and it's the way PatternFly does it.

The Required Actions solution uses chip groups, which are not appropriate here because chip groups get very hard to manage when you have more than a few items.  See the Login Events Settings for a really horrific example of this.

Reply:
The group selector is still not quite working right.  For instance, here there is no "s" in `/group1`, but it shows that group anyway.  Looks like it's because it has a subgroup with an "s" in it.
![group-select](https://user-images.githubusercontent.com/507202/93941008-8d8c6280-fcfb-11ea-9511-9fe6c41ad237.png)

Reply:
> @vmuzikar The `x` means "clear the field". That's different from remove and it's the way PatternFly does it.

@ssilvert Yes, I know what the "X" does but that might not be the case with the user. It might give the impression that the "X" removes the selection and therefore user might be unaware that this is a multi-select box.

> The Required Actions solution uses chip groups, which are not appropriate here because chip groups get very hard to manage when you have more than a few items. See the Login Events Settings for a really horrific example of this.

I assumed that no more that – let's say – 3 groups would be added here. It's still a manual process and IMHO doesn't make much sense for the admin to add tens of groups for just one user. But ok, the chip groups might not be the best solution.

I believe we should at least unify it with the Permissions UI – clear the search box after selection.

Reply:
> The group selector is still not quite working right. For instance, here there is no "s" in `/group1`, but it shows that group anyway. Looks like it's because it has a subgroup with an "s" in it.

This is what our REST API returns so I'd say it is out of scope of this PR. It's probably implemented like that because of the "tree" UI the Groups section uses. When you search for a group, you need to see all its parents to be able to properly display the tree.

Maybe we could do some filtering at the UI side but I'm not sure that's the correct approach.

Reply:
I've flatten and filtered out searched groups in latest commit. Also, I made sure the input is cleared after selection, so there's no confusion.